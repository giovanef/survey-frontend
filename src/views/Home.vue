<template>
  <div>
    <breadcrumbs :items="breadcrumbs"></breadcrumbs>

    <div class="pools">
      <div class="title">Enquetes</div>

      <div class="pools-utils">
        <router-link :to="{ name: 'pools-new' }" tag="button" class="btn btn-primary">
          Adicionar
        </router-link>
      </div>

      <template v-if="poolsLoading">
        <loading></loading>
      </template>
      <template v-else-if="poolsError">
        <error :error="poolsError"></error>
      </template>
      <template v-else>
        <pools-list :pools="pools"></pools-list>
      </template>
    </div>
  </div>
</template>

<script>
import api from '@/services/api';

import Breadcrumbs from '@/components/ui/Breadcrumbs.vue';
import Loading from '@/components/ui/Loading.vue';
import Error from '@/components/ui/Error.vue';
import PoolsList from '@/components/features/PoolsList.vue';

export default {
  data() {
    return {
      breadcrumbs: [
        { title: 'Home', to: { name: 'home' } },
      ],
      pools: [],
      poolsLoading: true,
      poolsError: false,
    };
  },
  mounted() {
    api.get('/pools')
      .then((response) => {
        this.poolsLoading = false;
        this.pools = response.data;
      })
      .catch((error) => {
        this.poolsError = error;
        this.poolsLoading = false;
      });
  },
  components: {
    Breadcrumbs,
    Loading,
    Error,
    PoolsList,
  },
};
</script>
