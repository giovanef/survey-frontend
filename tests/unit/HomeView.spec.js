import { shallowMount } from '@vue/test-utils';

import Home from '@/views/Home.vue';

describe('HomeView', () => {
  const build = () => {
    const wrapper = shallowMount(Home, {
      stubs: ['router-link'],
    });

    return {
      wrapper,
    };
  };

  beforeEach(() => {});

  it('renderiza o componente', () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
