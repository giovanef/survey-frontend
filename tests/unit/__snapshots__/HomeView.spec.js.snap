// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renderiza o componente 1`] = `
<div>
  <breadcrumbs-stub items="[object Object]"></breadcrumbs-stub>
  <div class="pools">
    <div class="title">Enquetes</div>
    <div class="pools-utils">
      <router-link-stub to="[object Object]" tag="button" class="btn btn-primary">
        Adicionar
      </router-link-stub>
    </div>
    <loading-stub></loading-stub>
  </div>
</div>
`;
